### PR TITLE
feat(windows): UI notification for reauth (#3329)

### PR DESCRIPTION
**Pull request by ReactorScram**
```[tasklist]
- [x] Smoke-test MSI from CI
- [x] Move out of drafts
- [x] Phrasing
```

### CODE REVIEW

File: rust/windows-client/src-tauri/src/client/gui.rs [hunk 1]
```diff
@@ -553,22 +571,14 @@ async fn run_controller(
...
-                        }
-                        controller.refresh_system_tray_menu()?;
+                    Req::CancelSignIn | Req::SignOut => {
+                        tracing::info!("User signed out or canceled sign-in");
+                        controller.sign_out()?;
+                    }
+                    Req::DisconnectedTokenExpired => {
+                        tracing::info!("Token expired");
+                        controller.sign_out()?;
+                        show_notification("Firezone disconnected", "To access resources, sign in again.")?;
```
Comment (jamilbk) [inline rust/windows-client/src-tauri/src/client/gui.rs:581]: 👍

**Review by jamilbk — APPROVED** [review]
LGTM

**Comment by jamilbk** [comment]
Refs #3329

File: rust/windows-client/src-tauri/src/client/gui.rs [hunk 2]
```diff
@@ -553,22 +571,14 @@ async fn run_controller(
...
-                            // the user canceled the sign-in or something innocent.
-                            tracing::warn!("tried to sign out but there's no session");
-                        }
-                        controller.refresh_system_tray_menu()?;
+                    Req::CancelSignIn | Req::SignOut => {
+                        tracing::info!("User signed out or canceled sign-in");
+                        controller.sign_out()?;
+                    }
+                    Req::DisconnectedTokenExpired => {
+                        tracing::info!("Token expired");
```
Comment (jamilbk) [inline rust/windows-client/src-tauri/src/client/gui.rs:579]: Just curious -- how are you detecting that the token has expired? IIRC it's an opaque token, so in most cases a `401` means the token has expired so this is probably fine

Comment (ReactorScram) [reply rust/windows-client/src-tauri/src/client/gui.rs:579]: I match on the error type in connlib's `on_disconnect` https://github.com/firezone/firezone/blob/e5e10b8641e609c57f2eb6688dcc33c445f47bc9/rust/windows-client/src-tauri/src/client/gui.rs#L311-L316

Looks like it was from #3142

**Pull request merged**